### PR TITLE
Add tooltips to RunAsync and CTAS button

### PR DESCRIPTION
**Pull request by vera-liu**
Issue:
 - [https://github.com/airbnb/superset/issues/1574](url)
 - [https://github.com/airbnb/superset/issues/1575](url)

Done:
 - added tooltips to Run Async and CTAS button

screen shot 2016-12-07 at 9 25 30 am
screen shot 2016-12-07 at 9 25 38 am

@ascott @bkyryliuk @mistercrunch

### CODE REVIEW

**Comment by mistercrunch** [comment]
Oh check out `assets/javascripts/components/Button.jsx` it's basically a bootstrap button with a tooltip prop.

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 1]
```diff
@@ -175,6 +179,7 @@ class SqlEditor extends React.PureComponent {
     }
     let ctasControls;
     if (this.props.database && this.props.database.allow_ctas) {
+      const ctasToolTip = 'run query with createTableAs';
```
Comment (bkyryliuk) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: s/ run query with createTableAs/Stores the results in the specified table

**Comment by bkyryliuk** [comment]
@jefffeng and @mistercrunch  - could u help with good phrasing?

**Comment by mistercrunch** [comment]
@elibrumbaugh !

**Comment by jefffeng** [comment]
Run Async = Run query asynchronously

CTAS = Create table as with query results

On Fri, Dec 9, 2016 at 11:34 AM Maxime Beauchemin <notifications@github.com>
wrote:

@elibrumbaugh <https://github.com/elibrumbaugh> !




—
You are receiving this because you were mentioned.


Reply to this email directly, view it on GitHub
<https://github.com/airbnb/superset/pull/1792#issuecomment-266101940>, or mute
the thread

**Comment by ascott** [comment]
i'm wondering about the user experience here, and thinking that we should only show a single run query button for simplicity. If run async is an available option, let's run the query async, if not we run a synchronous query, and in both cases we only show a singe `Run Query` button. thoughts?

**Comment by jefffeng** [comment]
@ascott - I really like the idea of having a single query button.  We shouldn't have to make users choose.  Rather, we should do the work of identifying what is the right way to run a query.  I'll put this as an item in our backlog

**Pull request merged**